dev/com.ibm.ws.jdbc_fat_derby/test-applications/jdbcapp/src/jdbc/fat/derby/web/JDBCDerbyServlet.java

```diff
@@ -10,37 +10,244 @@
+        InitialContext ctx = new InitialContext();
+
+        try {
+            DataSource ds4 = (DataSource) ctx.lookup("jdbc/dsfat4");
+            fail("Lookup should have failed due to bad config.");
+        } catch (Exception e) {
+            assertTrue("Exception message should have contained", e.getMessage().contains("CWWKN0008E"));
+        }
+    }
+
+    /**
+     * Data Sources - ds1 = DSConfig(TRAN_NONE) + No Res-ref + TRAN_NONE JDBC Driver
+     *
+     * Ensure connection is not enlisted in transaction
+     * Begin global transaction, do work, roll-back transaction, ensure work was not rolled-back.
+     */
+    @Test
+    public void testTransationEnlistment() throws Throwable {
+        createTable(ds1);
+        Connection con = null;
+
+        try {
+            con = ds1.getConnection();
+            tran.begin();
+            Statement stmt = con.createStatement();
+            stmt.executeUpdate("insert into cities values ('Rochester', 106769, 'Olmsted')");
+            ResultSet result = stmt.executeQuery("select county from cities where name='Rochester'");
+            if (!result.next())
+                throw new Exception("Entry missing from database");
+            String value = result.getString(1);
+            if (!"Olmsted".equals(value))
+                throw new Exception("Incorrect value: " + value);
+        } finally {
+            tran.rollback();
+            con.close();
+        }
+
+        try {
+            con = ds1.getConnection();
+            Statement stmt = con.createStatement();
+            ResultSet result = stmt.executeQuery("select county from cities where name='Rochester'");
+            if (!result.next()) {
+                throw new Exception("Entry missing from database after rollback. Connection should not have been enlisted in global transation.");
+            }
+            String value = result.getString(1);
+            if (!"Olmsted".equals(value))
+                throw new Exception("Incorrect value: " + value);
+        } finally {
+            con.close();
+        }
+    }
+
+    /**
+     * Data Sources - ds1 - DSConfig(TRAN_NONE) + No Res-ref + TRAN_NONE JDBC Driver
+     *
+     * Ensure we reject setAutoCommit(false)
+     */
+    @Test
+    @ExpectedFFDC({ "java.sql.SQLException" })
+    public void testRejectAutoCommit() throws Throwable {
+        createTable(ds1);
+        Connection con = null;
+
+        try {
+            con = ds1.getConnection();
+            tran.begin();
+            /*
+             * This should throw an exception because we are not enlisted in a global
+             * transaction (see previous testTransationEnlistment) so autoCommit(false)
+             * would prevent changes from being committed overall.
+             *
+             * If this behavior is undesired in the future consider a requirement to set
+             * CommitOrRollbackOnCleanup on dataSource.
+             */
+            con.setAutoCommit(false);
```

Comment:
You don't need all the code after this since you are expecting this call to result in an exception.